### PR TITLE
[No Ticket] Change queryparam passed to SHARE when filtering by resourceType

### PR DESCRIPTION
**Pull request by adlius**
-   Ticket: []
-   Feature flag: n/a

### CODE REVIEW

File: lib/osf-components/addon/components/search-page/component.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import uniqueId from 'ember-osf-web/utils/unique-id';
 
 interface ResourceTypeOption {
     display: string;
-    value: string;
+    value: string | null;
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/search-page/component.ts:21]: I think with this change, `@tracked resourceType: string;` should be updated to `@tracked resourceType?: string;`

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Nice! Did not cross my mind to use `@resourceType` instead of the defaultQP hash. I think the only other change we need is to update the `selectedResourceTypeOption` getter for when `this.resourceType === null`

File: lib/registries/addon/branded/discover/template.hbs [hunk 2]
```diff
@@ -7,6 +7,7 @@
     <SearchPage
         @provider={{this.model}}
         @defaultQueryOptions={{this.defaultQueryOptions}}
+        @resourceType={{'Registration,RegistrationComponent'}}
```
Comment (adlius) [inline lib/registries/addon/branded/discover/template.hbs:10]: If only we can import `enum`s in the template...

Comment (futa-ikeda) [reply lib/registries/addon/branded/discover/template.hbs:10]: Well, hypothetically we could if we wanted to try our first polaris component... ⭐

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**